spec/features/gardens/new_spec.rb

```diff
@@ -33,14 +27,42 @@
       expect(page).to have_button('Create Garden')
     end
 
-    it 'fills new garden form, submits and is redirect back to dashboard' do
+    it 'fills new garden form, submits and is redirected to garden index' do
+      name = 'The Grove'
+      longitude = 25.0000
+      latitude = 71.0000
+      description = 'My first garden'
+
+      expected_output = File.read('spec/fixtures/new_garden.json')
+
+      stub_request(:post, "https://solar-garden-be.herokuapp.com/api/v1/gardens?description=#{description}&latitude=#{latitude}&longitude=#{longitude}&name=#{name}&private=false&user_id=#{@user.id}").to_return(status: 200, body: expected_output, headers: {})
+
+      stub_request(:get, "https://solar-garden-be.herokuapp.com/api/v1/gardens/50").to_return(status: 200, body: expected_output, headers: {})
+
       visit new_garden_path
-      fill_in :name, with: 'Test'
-      fill_in :longitude, with: 25.0000
-      fill_in :latitude, with: 71.0000
+      fill_in :name, with: name
+      fill_in :longitude, with: longitude
+      fill_in :latitude, with: latitude
       fill_in :description, with: 'My first garden'
+      find('#private_false').click
+      
+      @user_with_garden = User.new({id: 1,
+      attributes: {
+        email: '123@gmail.com' },
+        relationships: {
+          gardens: {
+            data: [ {id: '50', type: 'garden'}] }}})
+            
+      allow_any_instance_of(ApplicationController).to receive(:current_user).and_return(@user_with_garden)
```

Comment:
Added this per Ian's suggestion because @user (the first user passed as current_user) doesn't actually get a new garden created from our test so we have to simulate it.